packages/database/src/core/ReadonlyRestClient.ts

```diff
@@ -78,7 +81,7 @@ export class ReadonlyRestClient extends ServerActions {
       c: boolean,
       d: number | null
     ) => void,
-    private authTokenProvider_: TokenProvider
+    private authTokenProvider_: AuthTokenProvider
```

Comment:
You used FirebaseAuthTokenProvider above.  I think they should both be AuthTokenProvider?